### PR TITLE
Clang-format and Clang-tidy

### PR DESCRIPTION
**Pull request by marcofavorito**
Apply clang-format and clang-tidy checks.

### CODE REVIEW

File: scripts/run-clang-format.py [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+#
+# Lydia is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Lydia.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""A wrapper script around clang-format, suitable for linting multiple files
```
Comment (marcofavorito) [inline scripts/run-clang-format.py]: taken from https://github.com/Sarcasm/run-clang-format

File: .clang-format [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+SpaceBeforeParens: ControlStatements
+SpaceBeforeRangeBasedForLoopColon: true
+SpaceInEmptyParentheses: false
+SpacesBeforeTrailingComments: 1
+SpacesInAngles:  false
+SpacesInContainerLiterals: true
+SpacesInCStyleCastParentheses: false
+SpacesInParentheses: false
+SpacesInSquareBrackets: false
+Standard:        Cpp11
```
Comment (francescofuggitti) [inline .clang-format]: shouldn't this be set to Cpp17?

Comment (marcofavorito) [reply .clang-format]: There's no such option in Clang 9.0:

https://releases.llvm.org/9.0.0/tools/clang/docs/ClangFormatStyleOptions.html

I think it is OK for now. If it raises issues in the future we will fix as they come.

**Comment by marcofavorito** [comment]
Fixes #10

File: .clang-format (hunk 2 above)
Comment (francescofuggitti) [reply .clang-format]: Ok, let's leave it for now

**Pull request merged**